Mathlib/Topology/CWComplex.lean

```diff
@@ -0,0 +1,140 @@
+/-
+Copyright (c) 2024 Elliot Dean Young and Jiazhen Xia. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiazhen Xia, Elliot Dean Young
+-/
+
+import Mathlib.Topology.Category.TopCat.Limits.Basic
+import Mathlib.Analysis.InnerProductSpace.PiL2
+import Mathlib.CategoryTheory.Functor.OfSequence
+
+/-!
+# CW-complexes
+
+This file defines (relative) CW-complexes.
+
+## Main definitions
+
+* `RelativeCWComplex`: A relative CW-complex is the colimit of an expanding sequence of subspaces
+`sk i` (called the `(i-1)`-skeleton) for `i ≥ 0`, where `sk 0` (i.e., the `(-1)`-skeleton) is an
+arbitrary topological space, and each `sk (n+1)` (i.e., the `n`-skeleton) is obtained from `sk n`
+(i.e., the `(n-1)`-skeleton) by attaching `n`-disks.
+
+* `CWComplex`: A CW-complex is a relative CW-complex whose `sk 0` (i.e., `(-1)`-skeleton) is empty.
+
+## References
+
+The definition of CW-complexes follows David Wärn's suggestion at
+https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Do.20we.20have.20CW.20complexes.3F/near/231769080
+-/
+
+open CategoryTheory
+
+universe u
+
+namespace RelativeCWComplex
+
+/-- The `n`-sphere is the set of points in ℝⁿ⁺¹ whose norm equals `1`,
+endowed with the subspace topology. -/
+noncomputable def sphere (n : ℤ) : TopCat.{u} :=
+  TopCat.of <| ULift <| Metric.sphere (0 : EuclideanSpace ℝ <| Fin <| Int.toNat <| n + 1) 1
+
+/-- The `n`-disk is the set of points in ℝⁿ whose norm is at most `1`,
+endowed with the subspace topology. -/
+noncomputable def disk (n : ℤ) : TopCat.{u} :=
+  TopCat.of <| ULift <| Metric.closedBall (0 : EuclideanSpace ℝ <| Fin <| Int.toNat n) 1
+
+/-- `𝕊 n` denotes the `n`-sphere. -/
+scoped notation "𝕊 "n => sphere n
+
+/-- `𝔻 n` denotes the `n`-disk. -/
+scoped notation "𝔻 "n => disk n
```

Comment:
...I think this should fix it
```suggestion
/-- `𝕊 n` denotes the `n`-sphere. -/
scoped notation "𝕊 "n:arg => sphere n

/-- `𝔻 n` denotes the `n`-disk. -/
scoped notation "𝔻 "n:arg => disk n
```

Reply:
Thank you, this works. (I also changed `notation` to `prefix` so that `n` can be omitted here.)